remote_attestation/java/src/com/google/oak/remote_attestation/crypto/AeadEncryptor.java

```diff
@@ -67,40 +68,40 @@ public AeadEncryptor(byte[] key) {
         // single encrypted field.
         byte[] encryptedData = encryptor.doFinal(data);
 
-        // Add `initializationVector` as a prefix to the `encryptedData`.
-        return Bytes.concat(initializationVector, encryptedData);
+        // Add `nonce` as a prefix to the `encryptedData`.
+        return EncryptedData.newBuilder()
+                .setNonce(ByteString.copyFrom(nonce))
+                .setData(ByteString.copyFrom(encryptedData))
+                .build();
     }
 
     /**
      * Decrypts and authenticates `data` using AES-GCM.
-     * `data` must contain an encrypted message prefixed with a 12 byte initialization vector.
      */
-    public byte[] decrypt(byte[] data) throws GeneralSecurityException {
+    public byte[] decrypt(EncryptedData data) throws GeneralSecurityException {
         Cipher decryptor = Cipher.getInstance(AEAD_ALGORITHM);
 
-        // Extract initialization vector from `data`.
-        byte[] initializationVector = Arrays.copyOf(data, INITIALIZATION_VECTOR_LENGTH_BYTES);
-        GCMParameterSpec gcmParameterSpecification = new GCMParameterSpec(TAG_LENGTH_BITS, initializationVector);
+        // Extract nonce from `data`.
```

Comment:
I don't think this comment is needed as extracting the nonce is no longer a complicated step.